frontend/src/components/file/FileUpload.vue

```diff
@@ -0,0 +1,94 @@
+<script setup lang="ts">
+import { storeToRefs } from 'pinia';
+import { ref } from 'vue';
+
+import { FileUpload } from '@/lib/primevue';
+import { documentService } from '@/services';
+import { useConfigStore } from '@/store';
+
+import type { FileUploadUploaderEvent } from 'primevue/fileupload';
+import type { Ref } from 'vue';
+
+// Props
+type Props = {
+  submissionId: string;
+};
+
+const props = withDefaults(defineProps<Props>(), {});
+
+const lastUploadedDocument = defineModel();
+
+// Store
+const { getConfig } = storeToRefs(useConfigStore());
+
+// State
+const fileInput: Ref<any> = ref(null);
+
+// Actions
+const onFileUploadDragAndDrop = (event: FileUploadUploaderEvent) => {
+  onUpload(Array.isArray(event.files) ? event.files[0] : event.files);
+};
+
+const onFileUploadClick = () => {
+  fileInput.value.click();
+};
+
+const onUpload = async (file: File) => {
+  lastUploadedDocument.value = (
```

Comment:
while this is concise, if the service call fails, setting the value of lastUploadedDocument to an error or rejected promise might not be ideal. not sure.. i guess it's ok.. i'm not sure on the best pattern but i guess it's good to catch the error where it first throws